numba/typed/typedlist.py

```diff
@@ -196,24 +204,47 @@ def empty_list(cls, item_type, allocated=DEFAULT_ALLOCATED):
         else:
             return cls(lsttype=ListType(item_type), allocated=allocated)
 
-    def __init__(self, **kwargs):
+    def __init__(self, *args, **kwargs):
         """
         For users, the constructor does not take any parameters.
         The keyword arguments are for internal use only.
 
         Parameters
         ----------
+        args: iterable
+            The iterable to intialize the list from
         lsttype : numba.core.types.ListType; keyword-only
             Used internally for the list type.
         meminfo : MemInfo; keyword-only
             Used internally to pass the MemInfo object when boxing.
         allocated: int; keyword-only
             Used internally to pre-allocate space for items
         """
+        illegal_kwargs = any((kw not in self._legal_kwargs for kw in kwargs))
+        if illegal_kwargs or args and kwargs:
+            raise TypeError("List() takes no keyword arguments")
         if kwargs:
             self._list_type, self._opaque = self._parse_arg(**kwargs)
         else:
             self._list_type = None
+            if args:
+                if not 0 <= len(args) <= 1:
+                    raise TypeError(
+                        "List() expected at most 1 argument, got {}"
+                        .format(len(args))
+                    )
+                iterable = args[0]
+                # Special case Numpy scalars or anything that quacks like a
+                # NumPy Array.
+                if hasattr(iterable, "ndim") and iterable.ndim == 0:
+                    self.append(iterable.item())
+                else:
+                    try:
+                        iter(iterable)
```

Comment:
I think every `Type` subclass will pass this check, because `Type` implements `__getitem__` to support `int32[:] -> array(int32, 1d, A)`.  I think this should be changed to `if isinstance(iterable, collections.abc.Iterable)`, which will only check for `__iter__`.

Reply:
o.k. can you think of a quick test to expose this brokenness?

Reply:
Nevermind, I read the code wrong yesterday. I thought the `iterable` would be instances of `Type`.